### ISSUE TITLE
Finetune ckpt inference fail

### ISSUE THREAD

**Issue by MingjieWe** [FoundationVision/InfinityStar#14]
亲爱的InfinityStar团队：
非常感谢你们的卓越工作。在复现过程中，我在微调后对于权重文件的保存和推理存在一些疑问，希望贵团队能帮助和指导，非常感谢！

具体情况如下：

我按照data/README和README进行了简单的1k的数据集的配置和训练。
在训练过程中，我保持了github中的参数，并且没有修改网络结构。在训练过程中Loss是正常的下降，Acc是正常的上升。
主要的修改集中在omnistore和torch的权重保存上，因为byte-omnistore的开源支持对FSDPCheckpointer存在bug，所以我采用了代码中的torch支持。

在推理过程中，不论是从随机初始化的权重或者从hugging face再次训练的模型，不论480p和720p的情况下，在推理时都有下图的情况。这也是这次issue希望能讨论和解决的问题。

如果有任何需要提供的信息，欢迎继续讨论～希望可以更好的复现工作和微调支持。再一次感谢贵团队的卓越工作！希望越来越好～

---------------------------------------------------------------------------------------------------------------------------------

Dear InfinityStar Team,

Thank you very much for your outstanding work. During the reproduction process, I encountered some questions regarding the saving of weights and inference after fine-tuning. I hope your team can provide guidance and assistance—thank you very much!

The situation is as follows:

I followed the configurations in data/README and README to set up and train a simple 1k dataset.
During training, I kept the parameters from GitHub and did not modify the network structure. The loss decreased normally, and the accuracy increased as expected.
The main modifications were focused on Omnistore and the PyTorch weight saving. Since the open-source support for Byte-Omnistore has a bug in FSDPCheckpointer, I used the PyTorch support provided in the code.

During inference, whether using randomly initialized weights or models further trained from Hugging Face, and regardless of 480p or 720p, the situation shown in the screenshot below occurs. This is the issue I hope to discuss and resolve.

If any further information is needed, I am happy to provide it. I hope this can help improve reproducibility and fine-tuning support. Once again, thank you for your outstanding work! Wishing your team continued success.

Faild 480p inference:
<img width="867" height="496" alt="Image" src="https://github.com/user-attachments/assets/666227f4-a0c0-4c56-b601-9caeec91c644" />

Faild 720p inference:
<img width="936" height="530" alt="Image" src="https://github.com/user-attachments/assets/decaf458-1fff-49be-a7ed-ebd27c69fada" />

The normal result with the original weights:
<img width="973" height="549" alt="Image" src="https://github.com/user-attachments/assets/fb06d32b-409d-44b7-9efb-8518cc3bab82" />

**Comment by happylicio** [comment]
@MingjieWe 
感谢你的关注，你能进一步debug，在关闭模型学习情况下，比如lr=0，训练几个step，然后保存ckpt推理看是否正常吗，我们可以先定位是ckpt保存问题，还是训练带来的问题

Thank you for your attention. Could you please further debug by disabling model learning? For example, set `lr=0`, train for a few steps, then save the checkpoint and run inference to check if it works as expected. This way, we can first determine whether the issue is related to checkpoint saving or to the training process itself.

**Comment by happylicio** [comment]
@MingjieWe 了解，也就是目前的问题是出在模型ckpt的保存上，或者推理load torch方式有bug，我们这边看下这个问题

**Comment by happylicio** [comment]
@MingjieWe 你有观察到采用torch load模型时候，打印里面有missing keys: [xxx]吗？

**Comment by happylicio** [comment]
@MingjieWe 我这边找到问题了，应该还是keys不匹配，你这边可以推理时候，尝试一下run_infinity.py里面infinity_test.load_state_dict(state_dict['trainer']['gpt_fsdp'])，保存时候没有只保存model，有一些其他meta信息，稍后我这边会修复下这个问题

**Comment by happylicio** [comment]
Thank you for your feedback. The bug has been fixed.

**Comment by MingjieWe** [comment]
Thanks for your help~
